interp/interp_test.go

```diff
@@ -1029,6 +1030,25 @@ func TestConfigVarsCorrect(t *testing.T) {
 	}
 }
 
+func TestShellCommand(t *testing.T) {
+	testGoAWK(t, `BEGIN { system("echo hello world") }`, "", "hello world\n", "", nil, nil)
+
+	if runtime.GOOS != "windows" {
+		testGoAWK(t, `BEGIN { system("world") }`, "", "hello world\n", "", nil,
+			func(config *interp.Config) {
+				config.ShellCommand = []string{"/bin/echo", "hello"}
+			})
+		testGoAWK(t, `BEGIN { "world" | getline; print }`, "", "hello world\n", "", nil,
+			func(config *interp.Config) {
+				config.ShellCommand = []string{"/bin/echo", "hello"}
+			})
+		testGoAWK(t, `BEGIN { print "hello world" | "-" }`, "", "hello world\n", "", nil,
+			func(config *interp.Config) {
+				config.ShellCommand = []string{"/bin/cat"}
+			})
+	}
```

Comment:
Nice, thanks for adding a test!

For a Windows test, it [looks like](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/cmd) we might be able to test with `config.ShellCommand = []string{"cmd.exe", "/c"}` ... though I'm not quite sure what would be a good, Windows-only command to run. Feel free to leave that out.